src/aggregation/runners/client-provider-distribution-acc.runner.ts

```diff
@@ -1,119 +1,75 @@
-                                                                     '1 week'::interval) dates)
-                                         select weeks.week                   as week,
-                                                client,
-                                                provider,
-                                                sum(total_deal_size)::bigint as total_deal_size,
-                                                sum(piece_size)::bigint      as unique_data_size
-                                         from weeks
-                                                  inner join miner_pieces
-                                                             on weeks.week >= miner_pieces.week
-                                         group by weeks.week,
-                                                  client,
-                                                  provider;`);
-
-        getDataEndTimerMetric();
-
-        const data: {
-          week: Date | null;
-          client: string | null;
-          provider: string | null;
-          total_deal_size: bigint | null;
-          unique_data_size: bigint | null;
-        }[] = [];
-
-        let isFirstInsert = true;
-
-        const storeDataEndTimerMetric =
-          startStoreDataTimerByRunnerNameMetric(runnerName);
-
-        for await (const rowResult of i) {
-          data.push({
-            week: rowResult.week,
-            client: rowResult.client,
-            provider: rowResult.provider,
-            total_deal_size: rowResult.total_deal_size,
-            unique_data_size: rowResult.unique_data_size,
-          });
-
-          if (data.length === 5000) {
-            if (isFirstInsert) {
-              await tx.$executeRaw`truncate client_provider_distribution_weekly_acc`;
-              isFirstInsert = false;
-            }
-
-            await tx.client_provider_distribution_weekly_acc.createMany({
-              data,
-            });
-
-            data.length = 0;
-          }
-        }
-
-        if (data.length > 0) {
-          if (isFirstInsert) {
-            await tx.$executeRaw`truncate client_provider_distribution_weekly_acc`;
-          }
-          await tx.client_provider_distribution_weekly_acc.createMany({
-            data,
-          });
-        }
-        storeDataEndTimerMetric();
-      },
-      {
-        timeout: Number.MAX_SAFE_INTEGER,
-      },
-    );
+    //const latestStored =
+    //  await prismaService.client_provider_distribution_weekly_acc.findFirst({
+    //    select: {
+    //      week: true,
+    //    },
+    //    orderBy: {
+    //      week: 'desc',
+    //    },
+    //  });
+    //let nextWeek = latestStored
+    //  ? DateTime.fromJSDate(latestStored.date, { zone: 'UTC' }) // we want to reprocess the last stored week, as it might've been incomplete
+    //  : DateTime.fromSeconds(3847920 * 30 + 1598306400).startOf('week'); // nv22 start week - a.k.a. reprocess everything
```

Comment:
This commented code is for incremental processing - it only recalculates since last processed week (inclusive, as last processed week could be incomplete or could be current ongoing week).

My plan is to first do the uncommented version that simply reprocesses everything every time. That's because we actually need to recalculate current data - it's wrong, data uniqueness was only calculated "per week" and not "up to a week". So I could store that same date on week 1 and week 2 and it wouldn't detect it as duplicates.

After it runs at least once, we can switch to the incremental version for performance.